pkg/fargate/fargate_test.go

```diff
@@ -156,7 +174,7 @@ var _ = Describe("fargate", func() {
 					Time: 0, TimeUnit: time.Second, MaxRetries: 5,
 				}
 				numRetriesBeforeDeletion := 3 // < MaxRetries
-				client := fargate.NewWithRetryPolicy(clusterName, mockForDeleteFargateProfileWithWait(profileName, numRetriesBeforeDeletion), retryPolicy)
+				client := fargate.NewWithRetryPolicy(clusterName, mockForDeleteFargateProfileWithWait(profileName, numRetriesBeforeDeletion), retryPolicy, nil)
```

Comment:
why can this be `nil`?

Reply:
because the stackManager only gets invoked when there are no other fargate stacks remaining, I added a "default" profile that gets returned in the `listFargateProfiles` by default below (line 420)

Reply:
I could replace all of the `nil`'s with a fake, but then we have to assert it wasn't called, which you get for free by passing nil in as it would panic the test

Reply:
can there be a test with the `stackManager` being called? I guess whenever I see `nil` in a test it always jumps out as like "chuck a nil in here, didn't write a test that covers that yet" otherwise it's not really an explicit expectation of the test.

Reply:
see line 146 :smile:

Reply:
ah, missed that. Maybe just defining like `neverCalledStackManager := nil` and then passing that in makes it clearer for people like me?

Reply:
Sure :+1:

Reply:
TIL you can't instantiate a varaible to nil directly, as it doesn't have a type :exploding_head: